### PR TITLE
Fix method (geocode or reverse) creation.

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

File: src/ProviderBase.php [hunk 1]
```diff
@@ -108,6 +108,9 @@ protected function process($method, array $data) {
       }
     }
 
+    // Extract the method properly.
+    list(, $method) = explode('::', $method, 2);
```
Comment (drupol) [inline src/ProviderBase.php]: If we don't do that, we end up with: CLASS::geocode or CLASS::reverse
And also an error in the log:

> 2088  15/Feb 07:04  warning  php   Warning: call_user_func_array() expects parameter 1 to be a valid callback, class 'Drupal\geocoder\Geocoder\Provider\Random' is not a subclass of 'Drupal\geocoder\ProviderBase' in Drupal\geocoder\ProviderBase->process() (line 113 of /home/pol/dev/git/geocoder/src/ProviderBase.php). 
>  2089  15/Feb 07:04  warning  geocoder  'Node one' cannot be geocoded.

**Comment by claudiu-cristea** [comment]
Oh, yes, you're right. I used `__METHOD__` but that's wrong. Then wouldn't be simpler to pass `__FUNCTION__` and we're done?

**Comment by drupol** [comment]
I haven't tried actually.

**Comment by claudiu-cristea** [comment]
Try it :)

**Comment by drupol** [comment]
Yep, will fix all your comment in the train in 1h from now :-)

**Comment by drupol** [comment]
Indeed, it works.

**Pull request merged**